cpp/src/reductions/scan.cu

```diff
@@ -33,37 +33,57 @@ struct ScanDispatcher {
     return std::is_arithmetic<T>::value;
   }
 
+  // @brief creates new column of same size, and copies null mask of input column view.
+  std::unique_ptr<column> allocate_like_copied_nullmask(const column_view& input,
```

Comment:
The awkward name of this function indicates to me that it is an awkward function. Instead, why not just call `allocate_like` to get a new column (with the option to not allocate the bit mask), and then call `col.set_null_mask(copy_bitmask(...), input.null_count())`?